sympy/simplify/simplify.py

```diff
@@ -2540,8 +2541,8 @@ def recurse(arg, **kwargs):
 
     expr = sympify(expr)
 
-    if not isinstance(expr, Basic) or expr.is_Atom or expr in (
-            exp_polar(0), exp_polar(1)):
+    if (not isinstance(expr, Basic) or isinstance(expr, MatrixSymbol) or (
+            expr.is_Atom or expr in (exp_polar(0), exp_polar(1)))):
```

Comment:
@jcrist : More analysis on the problem:

- When ```simplify``` calls ```powsimp```, the ```deep``` flag is turned on so in (https://github.com/sympy/sympy/blob/master/sympy/simplify/simplify.py#L2547-L2548), we recurse with the string name of the ```MatrixSymbol```
- In the next recursive call of ``powsimp```, the ```sympify``` line (https://github.com/sympy/sympy/blob/master/sympy/simplify/simplify.py#) converts the ```MatrixSymbol``` string name into a ```Symbol``` which leads to the crash when printing ```MatrixSymbol```

Using ```isinstance(..., str)``` is not as easy and obvious as in ```count_ops``` for this case. Therefore, I would opt for the solution of checking if an instance is ```MatrixSymbol``` (if so, stop processing further)